File: 189. Rotate Array/Solution.java
public class Solution {

    public void rotate(int[] nums, int k) {
        k %= nums.length;
        int length = nums.length - 1;
        reverseSpecificarray(nums, 0, length);
        reverseSpecificarray(nums, 0, k - 1);
        reverseSpecificarray(nums, k, length);
        

    }

    public void reverseSpecificarray(int[] arr, int start, int end) {
        int s = start;
        int e = end;
        while (s < e) {
            int temp = arr[s];
            arr[s] = arr[e];
            arr[e] = temp;
            s++;
            e--;
        }
    }

    public static void main(String[] args) {
        int[] array = { 1, 2, 3, 4, 5, 6, 7 };
        // Output: [5,6,7,1,2,3,4]
        int k = 3;
        Solution solution = new Solution();
        solution.rotate(array, k);
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }
}

File: 189. Rotate Array/Solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
    k%=nums.length;
  const length = nums.length - 1;
  reversearray(nums, 0, length);
  reversearray(nums, 0, k-1);
  reversearray(nums, k, length);
};

function reversearray(nums, start, end) {
  let s = start;
  let e = end;
  while (s < e) {
    temp = nums[s];
    nums[s] = nums[e];
    nums[e] = temp;
    s++;
    e--;
  }
}

const array = [1, 2, 3, 4, 5, 6, 7];
// Output: [5,6,7,1,2,3,4]
k = 3;
rotate(array,k)
console.log(array);
